test/integ_tests/hybrid_jobs/0_Creating_your_first_Hybrid_Job/0_Creating_your_first_Hybrid_Job_mocks.py

```diff
@@ -12,30 +13,60 @@ def pre_run_inject(mock_utils):
     mock_utils.mock_default_device_calls(mocker)
     mocker.set_search_result([
         {
-            "Roles" : [
+            "Roles": [
                 {
                     "RoleName": "AmazonBraketJobsExecutionRole",
-                    "Arn" : "TestRoleARN"
+                    "Arn": "TestRoleARN"
                 }
             ]
         }
     ])
-    mocker.set_create_job_result({
-        "jobArn" : f"arn:aws:braket:{mocker.region_name}:000000:job/testJob"
-    })
+    mocker.set_create_job_side_effect([
+        {
+            "jobArn": f"arn:aws:braket:{mocker.region_name}:000000:job/testJob"
+        },
+        {
+            "jobArn": f"arn:aws:braket:us-west-1:000000:job/testJob"
+        },
+    ])
     mocker.set_get_job_result({
-        "instanceConfig" : {
-            "instanceCount" : 1
+        "instanceConfig": {
+            "instanceCount": 1
         },
         "jobName": "testJob",
         "status": "COMPLETED",
         "outputDataConfig": {
-            "s3Path" : "s3://amazon-br-invalid-path/test-path/test-results"
+            "s3Path": "s3://amazon-br-invalid-path/test-path/test-results"
         }
     })
     mocker.set_log_streams_result({
         "logStreams": []
     })
+    mocker.set_get_query_results_result({
+        "status": "Complete",
+        "results": [
+            [
+                {"field": "@message", "value": "iteration_number=0;expval=0;"},
+                {"field": "@timestamp", "value": "0"},
+            ],
+        ]
+    })
+    mocker.set_batch_get_image_side_effect(
+        cycle([
+            {"images": [{"imageId": {"imageDigest": "my-digest"}}]},
+            {
+                "images": [
+                    {"imageId": {"imageTag": "my-tag"}},
+                    {"imageId": {"imageTag": "my-tag-py3"}},
+                    {"imageId": {"imageTag": f"my-tag-py3{sys.version_info.minor}-latest"}},
+                    {"imageId": {"imageTag": "latest"}},
+                ]
+            },
+        ])
+    )
```

Comment:
This seems unnecessarily complex, since this logic is/should be tested in the unit tests in the BDK.

Reply:
Updated. I still have alternating calls providing exclusively the digest and tag, since we know the tests should work with only those fields. If you think it's better to just simplify to one return value as you have, I'll go with that though